tests/cbmc/proofs/s2n_hmac_digest_verify/s2n_hmac_digest_verify_harness.c

```diff
@@ -0,0 +1,39 @@
+/*
+ * Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License").
+ * You may not use this file except in compliance with the License.
+ * A copy of the License is located at
+ *
+ *  http://aws.amazon.com/apache2.0
+ *
+ * or in the "license" file accompanying this file. This file is distributed
+ * on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+ * express or implied. See the License for the specific language governing
+ * permissions and limitations under the License.
+ */
+
+#include <assert.h>
+#include <cbmc_proof/proof_allocators.h>
+
+#include "crypto/s2n_hmac.h"
+
+void s2n_hmac_digest_verify_harness()
+{
+    /* Non-deterministic inputs. */
+    uint32_t len;
+    uint32_t alen;
+    uint32_t blen;
+    __CPROVER_assume(len <= MAX_ARR_LEN);
+    __CPROVER_assume(alen <= MAX_ARR_LEN);
+    __CPROVER_assume(blen <= MAX_ARR_LEN);
+    uint8_t *a = bounded_malloc(alen);
+    uint8_t *b = bounded_malloc(blen);
+
+    /* Check logical equivalence of s2n_constant_time_equals against element equality */
+    if (s2n_hmac_digest_verify(a, b, len) == S2N_SUCCESS) {
+        /* clang-format off */
+        assert(__CPROVER_forall { size_t i; (i < len) ==> (a[i] == b[i]) });
```

Comment:
I am somewhat unclear about this part.

`len`, `alen`, `blen` are all independently upper bounded by `MAX_ARR_LEN`. But if `alen` and `blen` are smaller than `len`, then wouldn't `a[i]` or `b[i]` fail?

Reply:
Good catch! I've updated this proof.